sklearn/ensemble/_forest.py

```diff
@@ -343,9 +348,19 @@ def fit(self, X, y, sample_weight=None):
         # Validate or convert input data
         if issparse(y):
             raise ValueError("sparse multilabel-indicator for y is not supported.")
+
+        feature_names_in_ = None
+        if isinstance(X, pd.DataFrame):
+            feature_names_in_ = X.columns.to_list()
```

Comment:
There is no need to extract the feature names here. When `_validate_data` is called, `self.feature_names_in_` will be set. This can be passed into `_parallel_build_trees` directly:

```diff
diff --git a/sklearn/ensemble/_forest.py b/sklearn/ensemble/_forest.py
index de729264e3..44be556ca4 100644
--- a/sklearn/ensemble/_forest.py
+++ b/sklearn/ensemble/_forest.py
@@ -467,6 +467,7 @@ class BaseForest(MultiOutputMixin, BaseEnsemble, metaclass=ABCMeta):
                     verbose=self.verbose,
                     class_weight=self.class_weight,
                     n_samples_bootstrap=n_samples_bootstrap,
+                    feature_names_in_=self.feature_names_in_,
                 )
                 for i, t in enumerate(trees)
             )
```

Reply:
Good one, let me try this.

Reply:
I tried your solution but a lot of the tests are failing 

```AttributeError: 'RandomForestClassifier' object has no attribute 'feature_names_in_'```

Reply:
Cheers, nice one. I implemented your suggestion and all tests are on green for `sklearn/ensamble/tests`.